charts/prometheus-node-exporter/templates/servicemonitor.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if .Values.prometheus.monitor.enabled }}
-apiVersion: monitoring.coreos.com/v1
+apiVersion: {{ .Values.prometheus.monitor.apiVersion | default "monitoring.coreos.com/v1" }}
```

Comment:
Imho this could be more generic. Instead if defining API version manually it could be checked which API version is available and the most recent one is used.

Reply:
I feel @lusson-luo intends to forcibly override apiVersion for matrix build purposes. Auto detect which apiVersion available is less concerned.

Reply:
> what api exactly is wanted? if its just version change it could be done automatically.
> 
> Something like:
> 
> `{{- if .Capabilities.APIVersions.Has "route.openshift.io/v1/Route" }}`

@monotek The scenario is like this, 
I encapsulate a node-exporter helm chart, which will be deployed on the k8s bases of various cloud manufacturers. The monitoring platforms of cloud manufacturers are different, but most of them will provide servicemonitors with promethues-like rules, but apiversion does not The same. When the operation and maintenance personnel deploy my charts, they can adapt only by modifying the values, without changing the helm source code.